ADVANCED_EXAMPLES.md

```diff
@@ -0,0 +1,409 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [`object.getStamp()`. 2 ways.](#objectgetstamp-2-ways)
+  - [First way](#first-way)
+  - [Second way](#second-way)
+- [Self cloneable objects. 3 ways.](#self-cloneable-objects-3-ways)
+  - [A composable stamp which adds `.clone()` to objects](#a-composable-stamp-which-adds-clone-to-objects)
+  - [Another way of self cloning](#another-way-of-self-cloning)
+  - [Memory efficient cloning](#memory-efficient-cloning)
+- [Delayed object instantiation using Promises](#delayed-object-instantiation-using-promises)
+- [Dependency injection tips](#dependency-injection-tips)
+- [Validate before a function call](#validate-before-a-function-call)
+- [EventEmitter without inheritance (`convertConstructor`)](#eventemitter-without-inheritance-convertconstructor)
+- [Hacking stamps](#hacking-stamps)
+  - ["Default" properties](#default-properties)
+  - ["Default" properties as composable behavior](#default-properties-as-composable-behavior)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+
+## `object.getStamp()`. 2 ways.
+
+> Run the examples below for yourself:
```

Comment:
I have to disagree on that.
1. Some might not even see the item `Run the Examples` in the table of contents. They might start reading from the `Memory efficient cloning`.
2. The `Run the Examples` is too broad. I wanted them to read example on this page, and then go to shell and launch the **specific** example. Not all of them.